synapse/config/logger.py

```diff
@@ -228,7 +231,25 @@ def _log(event):
             if event["log_text"].startswith("Timing out client"):
                 return
 
-        return observer(event)
+        # this is a workaround to make sure we don't get stack overflows when the
+        # logging system raises an error which is written to stderr which is redirected
+        # to the logging system, etc.
+        if getattr(threadlocal, "active", False):
+            # write the text of the event, if any, to the *real* stderr (which may
+            # be redirected to /dev/null, but there's not much we can do)
+            try:
+                event_text = eventAsText(event)
+                print("logging during logging: %s" % event_text, file=sys.__stderr__)
```

Comment:
I think this was meant to be "error during logging", but I'm not sure?

Reply:
well, we don't know for sure it was an error. It could have been any other log level.

Reply:
Hmm... "logging during logging" doesn't really make sense to me though. What's that trying to say?

Reply:
while we were writing some logging, more logging happened.

I'm open to better wording :)

Reply:
Ah, I see what you're saying. Well...if nothing else that should be easy to find in the code-base if it ever comes up. 😄